libs/cardano-ledger-api/test/Test/Cardano/Ledger/Api/State/QuerySpec.hs

```diff
@@ -238,62 +269,141 @@ propFilters ckFilter hkFilter statusFilter nes = do
     unless (Set.null statusFilter) $
       result `shouldSatisfy` const (allMemberStatuses `Set.isSubsetOf` statusFilter)
 
-genCommonMembers ::
-  Map.Map (Credential 'ColdCommitteeRole (EraCrypto era)) EpochNo ->
-  Map.Map
-    (Credential 'ColdCommitteeRole (EraCrypto era))
-    (Maybe (Credential 'HotCommitteeRole (EraCrypto era))) ->
+propNextEpoch ::
+  forall era.
+  EraGov era =>
+  NewEpochState era ->
+  Expectation
+propNextEpoch nes = do
+  withCommitteeInfo nes $
+    \(Committee cm _) (CommitteeState comStateMembers') nextComMembers noFilterResult -> do
+      let comMembers = Map.keysSet cm
+      let comStateMembers = Map.keysSet comStateMembers'
+
+      filterNext ToBeEnacted noFilterResult
+        `shouldSatisfy` (\res -> Map.keysSet res == nextComMembers `Set.difference` comMembers)
+
+      filterNext ToBeRemoved noFilterResult
+        `shouldSatisfy` (\res -> Map.keysSet res == (comMembers `Set.union` comStateMembers) `Set.difference` nextComMembers)
+
+      -- members who are both in current and nextCommittee are either ToBeExpired, TermAdjusted or NoChangeExpected
+      Set.unions
+        [ Map.keysSet (filterNext NoChangeExpected noFilterResult)
+        , Map.keysSet (filterNext ToBeExpired noFilterResult)
+        , Map.keysSet (termAdjusted noFilterResult)
+        ]
+        `shouldSatisfy` (== (comMembers `Set.intersection` nextComMembers))
+
+      let currentEpoch = csEpochNo noFilterResult
+      cmsExpiration
+        <$> filterNext ToBeExpired noFilterResult
+          `shouldSatisfy` all (all (== currentEpoch))
```

Comment:
There is a test failure on this line, which makes sense, since according to implementation we also need to look into `nextComMembers`.

```
Failures:

  test/Test/Cardano/Ledger/Api/State/QuerySpec.hs:300:11: 
  1) cardano-ledger-api.State.GetCommitteeMembersState CommitteeMembersState Query
       Falsifiable (after 7 tests and 1 shrink):
         fromList []
         Just (Committee {committeeMembers = fromList [(ScriptHashObj (ScriptHash "093103280a4ec6ee07112689e6ffe2af40a7e02f2f70bc4889851183"),EpochNo 8),(ScriptHashObj (ScriptHash "588a6d86d1699c0bd1edec2c92dc13a368255[831](https://github.com/input-output-hk/cardano-ledger/actions/runs/6975772642/job/18983342430?pr=3878#step:15:832)8c48acad2dc56917"),EpochNo 14),(ScriptHashObj (ScriptHash "9c3189b43d0cf6640fd666aa8fd22432d1897d737760b18d86ef57a7"),EpochNo 13),(KeyHashObj (KeyHash "cd1d95b25666eb69a049de36d8b61ff6ba073c6e2ba4109fa9c23bfe"),EpochNo 3)], committeeQuorum = 1 % 1})
         Just (Committee {committeeMembers = fromList [(ScriptHashObj (ScriptHash "093103280a4ec6ee07112689e6ffe2af40a7e02f2f70bc4889851183"),EpochNo 0),(ScriptHashObj (ScriptHash "35f78fe49efa041b892575913026746dc10c1e75ed3595aafe812653"),EpochNo 0),(ScriptHashObj (ScriptHash "588a6d86d1699c0bd1edec2c92dc13a3682558318c48acad2dc56917"),EpochNo 13)], committeeQuorum = 64862647 % 100000000})
         CommitteeState {csCommitteeCreds = fromList [(ScriptHashObj (ScriptHash "093103280a4ec6ee07112689e6ffe2af40a7e02f2f70bc4889851183"),Just (ScriptHashObj (ScriptHash "44b7363c58c9fe78b44a34782ae98cb1639cb95c2c98706ae2786875")))]}
         fromList [ScriptHashObj (ScriptHash "093103280a4ec6ee07112689e6ffe2af40a7e02f2f70bc4889851183"),ScriptHashObj (ScriptHash "20c5d88d990db9dfc6d23af8b27b232d56251edfa7d933b3051c9861"),ScriptHashObj (ScriptHash "588a6d86d1699c0bd1edec2c92dc13a3682558318c48acad2dc56917"),ScriptHashObj (ScriptHash "9c3189b43d0cf6640fd666aa8fd22432d1897d737760b18d86ef57a7"),ScriptHashObj (ScriptHash "b5d7ef53fe92a1eaef21d591bc1c03471070cf55eb4e108b384b59c8"),ScriptHashObj (ScriptHash "e813a73d6b45a99ff5ec56351190ce1b91e7a869e0a3c3e2cd3bcbcf"),KeyHashObj (KeyHash "4575131[835](https://github.com/input-output-hk/cardano-ledger/actions/runs/6975772642/job/18983342430?pr=3878#step:15:836)c8da45e54cbecf03c6fbd3fd6758b491f1c33823f747ff"),KeyHashObj (KeyHash "72dc08f4050aea7e1d26d66b18af06fc7405f639785be8ac16b04151"),KeyHashObj (KeyHash "baa3e30edd36077ae5023610888004789080f28f5bb645e84a739c19"),KeyHashObj (KeyHash "cd1d95b25666eb69a049de36d8b61ff6ba073c6e2ba4109fa9c23bfe")]
         fromList [ScriptHashObj (ScriptHash "138ab805795f19877b0feb86b282c476476d3060155b36d9d60ad499"),ScriptHashObj (ScriptHash "2a9c59edbed880d59976b97b78a93f9800542342b9adff5951e3c738"),ScriptHashObj (ScriptHash "d83d9677f130285887c0c94c503c72217e14938d917937469e21f9b3"),ScriptHashObj (ScriptHash "f2aa8f41c29b98354bc356ee38c57d76e2b8e5f77427a41f7ed51325"),KeyHashObj (KeyHash "278a398ba6e07a7f58db71302031ba69508d3a919331d447b9aa7a64")]
       predicate failed on: fromList [(ScriptHashObj (ScriptHash "093103280a4ec6ee07112689e6ffe2af40a7e02f2f70bc4889851183"),Just (EpochNo 8))]

  To rerun use: --match "/cardano-ledger-api/State/GetCommitteeMembersState/CommitteeMembersState Query/" --seed 581728073
```